### PR TITLE
Fix WebXR support for Oculus Browser

### PR DESCRIPTION
**Pull request by brianpeiris**
Oculus Browser has removed support for WebVR. This PR fixes our WebXR support and enables it for Oculus Browser specifically.
- Three.js' WebXRManager updates the scene's camera directly, instead of using the previous concept of a "poseTarget", so we need to grab headset transforms directly from the camera instead of the parent object3D
- Added a generic WebXRControllerDevice for WebXR gampads. The new webxr gamepad spec has different button and axis mappings, so we can't use existing controller devices, but the spec should now be much more standardized so the bindings should work for most devices when we expand webxr support. The bindings are a copy of our existing Oculus Touch bindings, with keyboard bindings removed and some cleanup. See https://immersive-web.github.io/webxr-gamepads-module/#xr-standard-gamepad-mapping
- Forced a max distance of 1000m on the skybox. Not sure why the camera's far field is different when entering webxr immersive mode. Perhaps something to investigate later.
- Updated anime.js to use `setTimeout` for animations instead of `window.requestAnimationFrame` since the global rAF is paused when in webxr immersive mode.
- Adds a `no_force_webvr` flag so that we can test webxr in other browsers.

Goes with:
- https://github.com/MozillaReality/aframe/pull/27
- https://github.com/MozillaReality/anime/pull/1

### CODE REVIEW

File: src/systems/waypoint-system.js [hunk 1]
```diff
@@ -365,10 +365,14 @@ export class WaypointSystem {
...
         elementFromTemplate.object3D.visible = this.scene.is("frozen");
         if (elementFromTemplate.object3D.visible) {
-          this.viewingCamera = this.viewingCamera || document.getElementById("viewing-camera").object3D;
+          this.viewingCamera = this.viewingCamera || document.getElementById("viewing-camera").object3DMap.camera;
           setMatrixWorld(
             elementFromTemplate.object3D,
-            calculateIconTransform(waypointComponent.el.object3D, this.viewingCamera, this.helperMat4)
+            calculateIconTransform(
+              waypointComponent.el.object3D,
+              this.viewingCamera.object3DMap.camera,
```
Comment (johnshaughnessy) [inline src/systems/waypoint-system.js:373]: This `object3DMap.camera` seems redundant with the `object3DMap.camera` above

File: src/systems/userinput/devices/webxr-controller.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import { paths } from "../paths";
+import { Pose } from "../pose";
+
+const ONES = new THREE.Vector3(1, 1, 1);
+const LEFT_HAND_OFFSET = new THREE.Matrix4().makeTranslation(-0.025, -0.03, 0.12);
```
Comment (johnshaughnessy) [inline src/systems/userinput/devices/webxr-controller.js:5]: Noting this for later -- these offsets should probably depend on the specific controller being used.

File: src/components/audio-feedback.js [hunk 3]
```diff
@@ -169,7 +169,7 @@ AFRAME.registerComponent("scale-audio-feedback", {
 
   async init() {
     await waitForDOMContentLoaded();
-    this.camera = document.getElementById("viewing-camera").object3D;
```
Comment (johnshaughnessy) [inline src/components/audio-feedback.js:172]: What was the motivation / reason for changing all these spots to use the camera object in the `object3DMap`?

Comment (johnshaughnessy) [reply src/components/audio-feedback.js:172]: Oops, nevermind I see you explained it in the summary!

**Comment by johnshaughnessy** [comment]
> Updated anime.js to use setTimeout for animations instead of window.requestAnimationFrame since the global rAF is paused when in webxr immersive mode.

This is alarming, because presumably the `timeout`s are more expensive than using a raf loop. What is the looping mechanism that replaces `window.requestAnimationFrame` for immersive webxr?

**Comment by johnshaughnessy** [comment]
> Three.js' WebXRManager updates the scene's camera directly, instead of using the previous concept of a "poseTarget", so we need to grab headset transforms directly from the camera instead of the parent object3D

Note for later: I'd like for us to control this ourselves because I think want to get the updated camera transform at the _beginning_ of the frame https://github.com/mozilla/hubs/issues/2229

File: src/systems/camera-system.js [hunk 4]
```diff
@@ -393,11 +388,11 @@ export class CameraSystem {
         this.avatarRig.object3D.updateMatrices();
         setMatrixWorld(this.viewingRig.object3D, this.avatarRig.object3D.matrixWorld);
         if (scene.is("vr-mode")) {
-          this.viewingCamera.object3D.updateMatrices();
-          setMatrixWorld(this.avatarPOV.object3D, this.viewingCamera.object3D.matrixWorld);
+          this.viewingCamera.object3DMap.camera.updateMatrices();
+          setMatrixWorld(this.avatarPOV.object3D, this.viewingCamera.object3DMap.camera.matrixWorld);
         } else {
           this.avatarPOV.object3D.updateMatrices();
-          setMatrixWorld(this.viewingCamera.object3D, this.avatarPOV.object3D.matrixWorld);
```
Comment (johnshaughnessy) [inline src/systems/camera-system.js:400]: I forget why there was this complexity around having the `viewingCamera` follow the `avatarPOV` node or vice versa, but I thought it was related to the `setPoseTarget` stuff and how it interacted with the-mode-formerly-known-as-inspect (aka focus mode). I'd just want to double check that focus and object list navigation works in 2D, touchscreen, and in VR.

File: src/systems/userinput/devices/webxr-controller.js (hunk 2 above)
Comment (brianpeiris) [reply src/systems/userinput/devices/webxr-controller.js:5]: Ah, good point. I'll add a comment to the code.

File: src/systems/waypoint-system.js (hunk 1 above)
Comment (brianpeiris) [reply src/systems/waypoint-system.js:373]: :+1: nice catch. I'll make sure to test the features I've touched as well.

**Comment by brianpeiris** [comment]
> This is alarming, because presumably the `timeout`s are more expensive than using a raf loop. What is the looping mechanism that replaces `window.requestAnimationFrame` for immersive webxr?

WebXR provides a requestAnimationFrame on the XRSession itself, which three.js and aframe use, but it would be hard to get anime.js to use it without some refactoring of the library. My understanding is that setTimeout isn't any more expensive -- it just uses the usual javascript event loop. Though I will try adding fixed timeout duration to the setTimeout, instead of the implicit 0ms timeout I'm using right now.

**Pull request merged**